### ISSUE TITLE
Wrong character encoding in German Version

### ISSUE THREAD

**Issue by Metibor** [Pyrdacor/Ambermoon.net#16]
When opening the inventory screen in the German version, we get an exception because of an encoding problem.
The UIText "MÄNNLICH" is read as "M─NNLICH" from the Executable.

Just changing `Encoding = Encoding.GetEncoding(437);` to another value in `Ambermoon.Data.Legacy.DataReader` does not work, because then the other data files won't load.

We could extend `CharToGlyph` with the additional characters though.

```
Unhandled exception. Ambermoon.AmbermoonException: [Data] Unsupported text character '─'.
   at Ambermoon.Data.Legacy.TextProcessor.CharToGlyph(Char ch, Boolean rune) in K:\GIT\Ambermoon.net\Ambermoon.Data.Legacy\Text.cs:line 117
   at Ambermoon.Data.Legacy.TextProcessor.<>c.<CreateText>b__1_0(Char ch) in K:\GIT\Ambermoon.net\Ambermoon.Data.Legacy\Text.cs:line 121
   at System.Linq.Enumerable.SelectEnumerableIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Ambermoon.Data.Legacy.TextProcessor.CreateText(String text) in K:\GIT\Ambermoon.net\Ambermoon.Data.Legacy\Text.cs:line 121
   at Ambermoon.UI.Layout.AddText(Rect rect, String text, TextColor color, TextAlign textAlign, Byte displayLayer) in K:\GIT\Ambermoon.net\Ambermoon.Core\UI\Layout.cs:line 393
   at Ambermoon.Game.<>c__DisplayClass92_0.<OpenPartyMember>b__0() in K:\GIT\Ambermoon.net\Ambermoon.Core\Game.cs:line 849
   at Ambermoon.Game.OpenPartyMember(Int32 slot) in K:\GIT\Ambermoon.net\Ambermoon.Core\Game.cs:line 870
   at Ambermoon.UI.Layout.Click(Position position, MouseButtons buttons, CursorType& cursorType) in K:\GIT\Ambermoon.net\Ambermoon.Core\UI\Layout.cs:line 614
   at Ambermoon.Game.OnMouseDown(Position position, MouseButtons buttons) in K:\GIT\Ambermoon.net\Ambermoon.Core\Game.cs:line 618
   at Ambermoon.GameWindow.Mouse_MouseDown(IMouse mouse, MouseButton button) in K:\GIT\Ambermoon.net\Ambermoon.net\GameWindow.cs:line 174
   at Silk.NET.Input.Desktop.GlfwMouse.HandleMouseDown(IMouse mouse, MouseButton button)
   at Silk.NET.Input.Desktop.GlfwMouse.<Subscribe>b__62_2(WindowHandle* _, MouseButton btn, InputAction action, KeyModifiers mods)
   at Silk.NET.Input.Desktop.GlfwEvents.<.ctor>b__0_2(WindowHandle* a, MouseButton b, InputAction c, KeyModifiers d)
   at Silk.NET.GLFW.XBD8D053E.Glfw.PollEvents()
   at Silk.NET.Windowing.Desktop.GlfwWindow.DoEvents()
   at Silk.NET.Windowing.Common.WindowExtensions.<>c__DisplayClass2_0.<Run>b__0()
   at Silk.NET.Windowing.Desktop.GlfwWindow.Run(Action onFrame)
   at Silk.NET.Windowing.Common.WindowExtensions.Run(IView view)
   at Ambermoon.GameWindow.Run(Configuration configuration) in K:\GIT\Ambermoon.net\Ambermoon.net\GameWindow.cs:line 250
   at Ambermoon.Program.Main() in K:\GIT\Ambermoon.net\Ambermoon.net\Program.cs:line 16
```

![grafik](https://user-images.githubusercontent.com/12674912/92108247-58b97980-ede7-11ea-8777-25c20fed0791.png)

DOS-Encoding
![grafik](https://user-images.githubusercontent.com/12674912/92108075-13954780-ede7-11ea-8277-f487b57b868e.png)

ANSI-Encoding
![grafik](https://user-images.githubusercontent.com/12674912/92108116-23149080-ede7-11ea-8b3e-3e458f1cd2cd.png)

**Comment by Pyrdacor** [comment]
Strange I thought the encoding worked. There is a glyph mapping inside the executable. I will add it tomorrow. I also found lots of UI graphics as well in the executable and will add them too. Today I have no time for it unfortunately.

**Comment by Pyrdacor** [comment]
Ah ok now I understand the problem. The data files use a custom encoding but the executable uses the Amiga string encoding which is Amiga Commodore and is nearly identical to ISO-Latin-1. I saw that there are multiple mappings between character codes and glyphs. Maybe I can make it work. Otherwise I will make an overload with custom encoding for the ReadString method of DataReader. The differences between Amiga Commodore and Latin 1 encodings shouldn't be a problem for any language that is supported by these kind of charsets.

**Comment by Pyrdacor** [comment]
@Metibor @Thallyrion Should be fixed in v1.0.18.
